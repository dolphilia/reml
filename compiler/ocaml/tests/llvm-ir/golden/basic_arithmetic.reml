// Basic arithmetic operations test (LLVM IR golden test)
//
// Tests:
// - Integer literals
// - let bindings
// - Function definitions
// - Binary operations (+, -, *, /)
// - Function calls

let x = 10
let y = 20

fn add(a, b) -> i64 = a + b

fn sub(a, b) -> i64 = a - b

fn mul(a, b) -> i64 = a * b

let sum = add(x, y)
let diff = sub(x, y)
let prod = mul(x, y)
