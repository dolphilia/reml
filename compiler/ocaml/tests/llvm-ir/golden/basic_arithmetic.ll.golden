; ModuleID = 'main'
source_filename = "main"
target datalayout = "e-m:e-p:64:64-f64:64:64-v128:128:128-a:0:64"
target triple = "x86_64-unknown-linux-gnu"

declare ptr @mem_alloc(i64)

declare void @inc_ref(ptr)

declare void @dec_ref(ptr)

; Function Attrs: noreturn
declare void @panic(ptr, i64) #0

define i64 @add(i64 %a, i64 %b) {
entry:
  %add_tmp = add i64 %a, %b
  ret i64 %add_tmp
}

define i64 @sub(i64 %a, i64 %b) {
entry:
  %sub_tmp = sub i64 %a, %b
  ret i64 %sub_tmp
}

define i64 @mul(i64 %a, i64 %b) {
entry:
  %mul_tmp = mul i64 %a, %b
  ret i64 %mul_tmp
}

attributes #0 = { noreturn }
